enterprise/app/filter/filter.tsx

```diff
@@ -55,7 +69,30 @@ type CustomDateRange = Range & {
 const LAST_N_DAYS_OPTIONS = [7, 30, 90, 180, 365];
 
 export default class FilterComponent extends React.Component<FilterProps, State> {
-  state: State = {};
+  state: State = this.getState();
+
+  componentDidUpdate(prevProps: FilterProps) {
+    if (this.props.search != prevProps.search) {
+      this.setState(this.getState());
+    }
+  }
+
+  getState() {
```

Comment:
nit: might be slightly cleaner to refactor this as `advancedFilterStateFromUrl(search: URLSearchParams)` (I think TS will let you pass in `this.props.search` when calling this from the top-level of the class)

Reply:
Done